From 59cb9f0d043b9cc6726500ed9af3dff946784475 Mon Sep 17 00:00:00 2001
From: Utkarsh Ayachit <utkarsh.ayachit@kitware.com>
Date: Thu, 16 Jan 2014 17:31:34 -0500
Subject: [PATCH] Cleanup dependencies on some core modules.

Cleaned up how dependencies on some of the core VTK modules was added.
Also removed the need for vtktiff in VisItBridge. We don't need the tiff writer
in this plugin. It was causing build issues when using system tiff.
---
 Library/VisItLib/CMake/VisItCommon.cmake       |   26 ------------------------
 Library/VisItLib/visit_vtk/full/CMakeLists.txt |    3 +--
 Library/module.cmake                           |    6 ++++--
 3 files changed, 5 insertions(+), 30 deletions(-)

diff --git a/Library/VisItLib/CMake/VisItCommon.cmake b/Library/VisItLib/CMake/VisItCommon.cmake
index 30384e3..70e82c6 100644
--- a/Library/VisItLib/CMake/VisItCommon.cmake
+++ b/Library/VisItLib/CMake/VisItCommon.cmake
@@ -86,19 +86,6 @@ FUNCTION(ADD_PARALLEL_LIBRARY target)
     ENDIF(VISIT_PARALLEL_CXXFLAGS)
 ENDFUNCTION(ADD_PARALLEL_LIBRARY)
 
-MACRO(VISIT_VTK_THIRD_PARTY_INCLUDE upper lower)
-  if(VTK_USE_SYSTEM_${upper})
-    if(${upper}_INCLUDE_DIR)
-      include_directories(${${upper}_INCLUDE_DIR})
-    endif(${upper}_INCLUDE_DIR)
-  else(VTK_USE_SYSTEM_${upper})
-    include_directories(
-      ${VTK_BINARY_DIR}/Utilities/${lower}
-      ${VTK_SOURCE_DIR}/Utilities/${lower}
-    )
-  endif(VTK_USE_SYSTEM_${upper})
-ENDMACRO(VISIT_VTK_THIRD_PARTY_INCLUDE)
-
 #called from readers that are being built into paraview
 FUNCTION(ADD_VISIT_READER NAME VERSION)
   set(PLUGIN_NAME "vtk${NAME}")
@@ -612,19 +599,6 @@ if(MILI_FOUND)
   set(HAVE_LIBMILI ${MILI_FOUND})
 endif(MILI_FOUND)
 
-#setup zlib
-include_directories(${VTK_ZLIB_INCLUDE_DIRS})
-
-#setup netcdf
-visit_vtk_third_party_include(NETCDF vtknetcdf/include)
-
-#setup png
-visit_vtk_third_party_include(PNG vtkpng)
-
-#setup tiff
-visit_vtk_third_party_include(TIFF vtktiff)
-
-
 #-----------------------------------------------------------------------------
 # Detect packages here. We could probably write macros that we can include from
 # elsewhere for this.
diff --git a/Library/VisItLib/visit_vtk/full/CMakeLists.txt b/Library/VisItLib/visit_vtk/full/CMakeLists.txt
index babb836..fa46408 100644
--- a/Library/VisItLib/visit_vtk/full/CMakeLists.txt
+++ b/Library/VisItLib/visit_vtk/full/CMakeLists.txt
@@ -88,7 +88,7 @@ vtkVisItSplitter.C
 vtkVisItStreamLine.C
 vtkVisItStructuredGrid.C
 vtkVisItStructuredGridNormals.C
-vtkVisItTIFFWriter.C
+#vtkVisItTIFFWriter.C
 vtkVisItTensorGlyph.C
 vtkVolumeFromVolume.C
 )
@@ -130,7 +130,6 @@ target_link_libraries(visit_vtk
     vtkInteractionStyle
     vtkIOLegacy
     vtkRenderingOpenGL
-    vtktiff
   )
 
 VISIT_INSTALL_TARGETS(visit_vtk)
diff --git a/Library/module.cmake b/Library/module.cmake
index ef4a07f..91629a6 100644
--- a/Library/module.cmake
+++ b/Library/module.cmake
@@ -3,14 +3,16 @@ vtk_module(VisItLib
     vtkCommonDataModel
     vtkCommonExecutionModel
     vtkCommonMisc
-    vtkFiltersExtraction
     vtkFiltersAMR
+    vtkFiltersExtraction
     vtkFiltersFlowPaths
     vtkImagingHybrid
     vtkIOGeometry
     vtkIOImage
+    vtkIOLegacy
     vtkParallelCore
+    vtkpng
     vtkRenderingVolumeOpenGL
-    vtkIOLegacy
+    vtkzlib
   EXCLUDE_FROM_WRAPPING
   )
-- 
1.7.10.4

